sky/execution.py

```diff
@@ -628,18 +628,22 @@ def spot_launch(
                 'and comment out the task names (so that they will be auto-'
                 'generated) .')
         task_names.add(task_.name)
-    for task_ in dag.tasks:
-        assert len(task_.resources) == 1, task_
-        resources = list(task_.resources)[0]
 
-        change_default_value: Dict[str, Any] = {}
-        if not resources.use_spot_specified:
-            change_default_value['use_spot'] = True
-        if resources.spot_recovery is None:
-            change_default_value['spot_recovery'] = spot.SPOT_DEFAULT_STRATEGY
+    for task_ in dag.tasks:
 
-        new_resources = resources.copy(**change_default_value)
-        task_.set_resources({new_resources})
+        new_resources_list = []
+        for res in task_.get_resources_list():
+
+            override_params: Dict[str, Any] = {}
+            if not task_.is_resources_mixed_spot:
```

Comment:
should we raise error if resources are mixed with spot & on-demand?

Reply:
I will remove this part and defer it to a later PR>